app/lib/publisher/backend.dart

```diff
@@ -329,3 +319,29 @@ class PublisherBackend {
 
 api.PublisherInfo _asPublisherInfo(Publisher p) =>
     api.PublisherInfo(description: p.description, contactEmail: p.contactEmail);
+
+/// Loads [publisherId] and checks if [userId] is an admin of the publisher, and
+/// runs the callback [fn] with it.
+///
+/// Throws AuthenticationException if the user is provided.
+/// Throws AuthorizationException if the user is not an admin for the publisher.
+Future<R> withPublisherAdmin<R>(
+    String publisherId, String userId, FutureOr<R> fn(Publisher p)) async {
+  if (userId == null) {
```

Comment:
nit: use `ArgumentError.checkNotNull(userId, 'userId');` since this is just an assertion. Callers are supposed to check authentication first.